app/classes/textile.rb

```diff
@@ -362,6 +399,38 @@ def pre_existing_links_replaced_by_placeholders!(do_object_links)
     saved_links
   end
 
+  # rubocop:disable Style/RegexpLiteral
+  # cop gives false positive
+  IMPLICIT_TERM_PATTERN = /
+    (?<! x{NAME) # discard match if preceded by an MO object tag
+    (?<! x{GLOSSARY_TERM)
+    (?<! x{OBSERVATION)
+    (?<! x{LOCATION)
+    (?<! x{USER)
+    (?<! x{IMAGE)
+    (?<! x{PROJECT)
+    (?<! x{SPECIES_LIST)
+    (?<! x{COMMENT)
```

Comment:
@pellaea, @mo-nathan 
### Is there a better solution?
These negative lookbehinds prevent further munging of MO tagged objects.
Maybe supplement the negative lookahead at the end of the regexp?

Reply:
I don't think I understand the kinds of cases this guards against.  I think it would stop the second link in this:

```
Blah _observation 123_ _glossary term_ blah.
```
and that doesn't seem right.  What is the intention?

Reply:
Fair question. Sorry I didn't explain it better.

TLDR: The intent is to prevent tagged objects from being munged.

The regexp `IMPLICIT_TERM_PATTERN` is used by the `convert_implicit_terms_to_tagged_glossary_terms!` method.
The purpose of that method is to convert implicit glossary links (ex: `_lichen_`).

When the method is called, `self` has been modified. 
In particular, the explicit object links (ex: `_user 252_`) and implicit Name links (ex: `_Lichen_`) have been converted to tagged objects.

Each tagged object takes the form 
`<prefix>x{OBJECT_NAME __label__ }{ id }x`
So each tagged object includes a substring within underscores.

I want `convert_implicit_terms_to_tagged_glossary_terms!` to leave the tagged objects alone -- _not_ convert that substring to a tagged GlossaryTerm object.

Does that make sense?

Reply:
Oh my gosh, I get it now.  I was fooled by the `{x` business.  I never really learned all the fancy new regex notations and thought that was one... but no, it's literal. :)

Right, so I see why it's necessary.  I do think it *may* be better to do a single negative look-behind assertion instead of many:

```
(?<! (?:FOO|BAR|BAZ|etc.) )
```

Maybe.  Also, just a thought... how often do people include literal words like "x{XXXXX" in their mark-up???  Probably not often enough to worry.  In which case this would definitely be shorter/easier (if not as precise):

```
(?<! x{[A-Z_]+)
```

Reply:
Thanks again Jason. I appreciate your time and thoughts.

-  `(?<! x{[A-Z_]+)` is a good idea. But it throws `invalid pattern in look-behind: / (SyntaxError)`.  
The problem is that Ruby does not allow variable length lookbehinds. https://learnbyexample.github.io/Ruby_Regexp/lookarounds.html#variable-length-lookbehind  
-  `(?<! (?:FOO|BAR|BAZ|etc.) )`: I need to think about this. (I've been fighting regexp's for a solid week and am exhausted.)
(I probably can make something similar work because Ruby does allow different length alternatives in lookbehinds, provided that each alternative is fixed length. But my understanding is that will not improve efficiency because "Each alternative is treated as a separate fixed-length lookbehind." https://www.regular-expressions.info/lookaround.html

Reply:
Your week of studying regexen has really paid off!  You have officially surpassed me.  I used to be so good with them.  You've convinced me that your solution is the best available.  (At least without a major refactor.)